src/load/azext_load/data_plane/load_test/custom.py

```diff
@@ -302,6 +307,93 @@ def convert_to_jmx(
 
 
+def set_baseline(
+    cmd,
+    load_test_resource,
+    test_id,
+    test_run_id,
+    resource_group_name=None,
+):
+    logger.info("Setting baseline for test with test ID: %s", test_id)
+    test_client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    existing_test = test_client.get_test(test_id)
+    existing_test_run = test_run_client.get_test_run(test_run_id)
+    if existing_test_run.get("testId") != test_id:
+        raise InvalidArgumentValueError(
+            f"Test run with ID: {test_run_id} is not associated with test ID: {test_id}"
+        )
+    if existing_test_run.get("status") not in ["CANCELLED", "DONE"]:
+        raise InvalidArgumentValueError(
+            f"Test run with ID: {test_run_id} does not have a valid "
+            f"test run status {existing_test_run.get('status')}. "
+            "Valid test run status are: CANCELLED, DONE"
+        )
+    if existing_test_run.get("testRunStatistics", {}).get("Total") is None:
+        raise InvalidArgumentValueError(
+            f"Sampler statistics are not yet available for test run ID {test_run_id}. "
+            "Please try again later."
+        )
+    body = create_or_update_test_without_config(
+        test_id=test_id,
+        body=existing_test,
+        baseline_test_run_id=test_run_id,
+    )
+    response = test_client.create_or_update_test(test_id=test_id, body=body)
+    logger.debug("Set test run %s as baseline for test: %s", test_run_id, test_id)
+    return response.as_dict()
+
+
+def compare_to_baseline(
+    cmd,
+    load_test_resource,
+    test_id,
+    resource_group_name=None,
+    response_time_aggregate=AllowedTrendsResponseTimeAggregations.MEAN.value,
+):
+    logger.info("Showing test trends for test with test ID: %s", test_id)
+    test_client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test = test_client.get_test(test_id)
+    if test.get("baselineTestRunId") is None:
+        raise InvalidArgumentValueError(
+            f"Test with ID: {test_id} does not have a baseline test run associated with it."
+        )
+    baseline_test_run_id = test.get("baselineTestRunId")
+    baseline_test_run = test_run_client.get_test_run(baseline_test_run_id)
+    all_test_runs = test_run_client.list_test_runs(test_id=test_id)
+    all_test_runs = [run.as_dict() for run in all_test_runs]
+    logger.debug("Total number of test runs: %s", len(all_test_runs))
+    count = 0
+    recent_test_runs = []
+    for run in all_test_runs:
+        if (
+            run.get("testRunId") != baseline_test_run_id
+            and count < 10  # Show only 10 most recent test runs
+            and run.get("status") in ["CANCELLED", "DONE"]
+        ):
+            recent_test_runs.append(run)
```

Comment:
I tried that earlier, but the status is not getting filtered properly as I noticed that with the unit test when it's CANCELLED status and maxPageSize is not available with 2024-05 preview sdk

Reply:
Ensure the values passed are correct. Filtering should work server side, otherwise its a bug in the system no? Status is a comma separated list of values fwiw

Reply:
yes, it's a bug, from the comma separated list only the first list item status is getting honored, so I can't depend on API just yet and since I don't know on what page CANCELLED and DONE test runs exist, using maxPageSize seems redundant

Reply:
when I set `test_run_client.list_test_runs(orderby="executedDateTime desc",test_id=test_id,status="CANCELLED, DONE")`
![image](https://github.com/user-attachments/assets/8353552c-a906-49a7-92ce-734f1b2488db)
when I set `test_run_client.list_test_runs(orderby="executedDateTime desc",test_id=test_id,status="DONE,CANCELLED")`
![image](https://github.com/user-attachments/assets/fa585445-0ea5-4424-b333-7cf70ab0b2fb)

The first one in the list is baseline test run, that will come either ways

Reply:
I did manual testing on the API, looks like the bug is not on API but in SDK, how can I test SDK?

Reply:
nvm, the issue is arising when I use space in the list like `CANCELLED, DONE`, working fine with `CANCELLED,DONE`. weird but it is what it is 😂

Reply:
Yes, lol. I dont think the API handles spacing